### PR TITLE
Make Mt-KaHyPar compatible with OneTBB (fix #106)

### LINKED ISSUE #106: Problems with TBB 2021 Versions
The releases of TBB from 2021 no longer contain some headers we use:
* `tbb_stddef.h` which is read to determine the interface version with CMake. This information was moved to https://github.com/oneapi-src/oneTBB/blob/master/include/oneapi/tbb/version.h
* `task_scheduler_init.h` became deprecated in favor of this (?) https://github.com/oneapi-src/oneTBB/blob/master/include/oneapi/tbb/global_control.h

For now the Ubuntu Repos all provide the 2020 Versions of TBB, but rolling release distros already use the 2021 version, which Ubuntu will eventually use, too.

### PR DESCRIPTION
**Pull request by kittobi1992**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -125,12 +122,6 @@ else()
   endif()
 endif()
 
-if(KAHYPAR_USE_COMPATIBLE_TBB_VERSION)
```
Comment (larsgottesbueren) [inline CMakeLists.txt:128]: This was actually a nice feature, that we may want to keep around?

File: mt-kahypar/parallel/stl/zero_allocator.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  using propagate_on_container_move_assignment = std::true_type;
+  using is_always_equal = std::true_type;
+
+  zero_allocator() = default;
+  template <typename U>
+  explicit zero_allocator(const zero_allocator<U>&) noexcept {}
+
+  T* allocate(std::size_t n) {
+    T* ptr = tbb::tbb_allocator<T>::allocate(n);
+    std::memset(static_cast<void*>(ptr), 0, n * sizeof(value_type));
```
Comment (larsgottesbueren) [inline mt-kahypar/parallel/stl/zero_allocator.h:47]: do we want to parallelize this?

Comment (kittobi1992) [reply mt-kahypar/parallel/stl/zero_allocator.h:47]: actually copies the previous implementation of tbb::zero_allocator

File: CMakeLists.txt (hunk 1 above)
Comment (kittobi1992) [reply CMakeLists.txt:128]: could be an idea for environment where you not allowed to install something

**Pull request merged**